eng/pipelines/templates/stages/1es-redirect.yml

```diff
@@ -11,7 +11,7 @@ resources:
     - repository: azure-sdk-build-tools
       type: git
       name: internal/azure-sdk-build-tools
-      ref: refs/tags/azure-sdk-build-tools_20250228.1
+      ref: refs/heads/users/pahallis/credscan
```

Comment:
[FYI for reviewers](https://dev.azure.com/azure-sdk/internal/_git/azure-sdk-build-tools/pullrequest/1156)

Reply:
Does the credscansuppression file coming down as part of azure-sdk-build-tools put the credscan suppression file in the same place as exists in the repo? I would expect it to exist under a nonstandard path `$(Pipeline.Workspace)/azure-sdk-build-tools/eng/CredScanSuppression.json`.

Reply:
Path variables aren't consistent when mixed with different `checkout` step combinations.

When a run has a single `checkout` step, `$(System.DefaultWorkingDirectory)` always points to the single repo path, even when the repo uses a custom path.

When there are multiple `checkout` steps, `$(System.DefaultWorkingDirectory)` always points to `$(Pipeline.Workspace)/s`

The credscan path seems to be rooted with `$(System.DefaultWorkingDirectory)`, so it should track single repo checkouts, but the sdl settings will need to be overriden for any multi-repo checkout.

I added a wiki page that covers how the path variables are affected by different repo counts and paths:
https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/1222/Pipeline-Path-Variables